### PR TITLE
How to mock in testing section of contributing guide

### LINKED ISSUE #107: Update CONTRIBUTING with information on unit testing and mocking
- [ ] Emphasize the value of unit testing over functional tests.
- [ ] Emphasize the value of mocking in unit tests.
- [ ] Outline common / esoteric mocking examples.

### PR DESCRIPTION
**Pull request by FyzHsn**
Fixes #107

### CODE REVIEW

**Comment by FyzHsn** [comment]
@wjcunningham7 @kessler-frost Just updated the section on testing: guiding principles of writing tests, what to mock, the essential commands useful for mocking and also some relevant stack overflow discussions / documentation link.

This is a pretty small review.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -385,43 +385,70 @@ export LOGLEVEL=INFO
...
+
+The Covalent test suite can be run locally from the root directory with `pytest` (add the tags `-vv -s` for a more verbose setting).
 
-Running the entire test suite will take a while and for the purpose of development, one can
-focus on more specific tests using syntax in the examples shown below.
+Running the entire test suite everytime the code has been updated will take a while. The following commands are useful to hone in on a smaller subset of tests for development purposes.
 
 1. To run a specific test module:
-```buildoutcfg
+```python
```
Comment (kessler-frost) [inline CONTRIBUTING.md:414]: Hmm, this is not python code though 😅 .

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -385,43 +385,70 @@ export LOGLEVEL=INFO
...
 pytest tests/covalent_dispatcher_tests/_executor/local_test.py -vv -s
 ```
 
 2. To run a specific test:
 
-```buildoutcfg
+```python
```
Comment (kessler-frost) [inline CONTRIBUTING.md:420]: Neither is this 😶 .

**Review by kessler-frost — COMMENTED** [review]
Minor comments, otherwise it's gtg.

**Review by kessler-frost — APPROVED** [review]

**Pull request merged**